services/spar/src/Spar/Scim.hs

```diff
@@ -108,8 +107,45 @@ apiScim = hoistScim (toServant (Scim.siteServer configuration))
      :<|> apiScimToken
   where
     hoistScim = hoistServer (Proxy @(Scim.SiteAPI ScimToken))
-                            (Scim.fromScimHandler fromError)
-    fromError = throwError . SAML.CustomServant . Scim.scimToServantErr
+                            (wrapScimErrors . toSpar)
+
+    -- Unwrap the 'ScimHandler'
+    toSpar :: Scim.ScimHandler Spar a -> Spar a
+    toSpar = Scim.fromScimHandler
+             (throwError . SAML.CustomServant . Scim.scimToServantErr)
+
+    -- Wrap all errors into the format required by SCIM.
+    --
+    -- FIXME: this doesn't catch impure exceptions (e.g. thrown with 'error').
+    -- Let's hope that SCIM clients can handle non-SCIM-formatted errors
+    -- properly. See <https://github.com/haskell-servant/servant/issues/1022>
+    -- for why it's hard to catch impure ex
```

Comment:
I wouldn't call this a `FIXME`, but rather a `NB:`.  Just don't use impure exceptions.  But either way it certainly is worth mentioning this here.